### PR TITLE
[itay-grudev-singleapplication] Add new port

### PR DESCRIPTION
**Pull request by nickdademo**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [ ] ~~Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)~~
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [ ] ~~Only one version is added to each modified port's versions file.~~

### CODE REVIEW

**Comment by nickdademo** [comment]
NOTE: Project is named `singleapplication` at https://repology.org/project/singleapplication/versions. This port uses the name `itay-grudev-singleapplication` as advised [here](https://github.com/microsoft/vcpkg/issues/27653#issuecomment-1805081948)

File: ports/itay-grudev-singleapplication/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+if("qguiapplication" IN_LIST FEATURES)
+    set(QAPPLICATION_CLASS QGuiApplication)
+elseif("qapplication" IN_LIST FEATURES)
+    set(QAPPLICATION_CLASS QApplication)
+elseif("qcoreapplication" IN_LIST FEATURES)
+    set(QAPPLICATION_CLASS QCoreApplication)
+else()
+    set(QAPPLICATION_CLASS QCoreApplication)
+endif()
```
Comment (dg0yt) [inline ports/itay-grudev-singleapplication/portfile.cmake:24]: I do understand that this lib is intentended to be used by executables, not libraries.
But I'm afraid this kind of using features to select alternatives is not acceptable for vcpkg.
https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#do-not-use-features-to-implement-alternatives

Comment (nickdademo) [reply ports/itay-grudev-singleapplication/portfile.cmake]: Overlooked this. Thanks! I have removed the features and now just set `QAPPLICATION_CLASS=QGuiApplication` in the port (as this is probably the most widely used config).

**Review by LilyWangLL — APPROVED** [review]

**Comment by LilyWangLL** [comment]
Usage test passed on `x64-windows`.

**Comment by data-queue** [comment]
I'm not very familiar with Qt, so I'm not exactly sure what this library is for. Is it to replace a Qt library?

**Comment by nickdademo** [comment]
> I'm not very familiar with Qt, so I'm not exactly sure what this library is for. Is it to replace a Qt library?

I think around 10+ years ago, Qt offered this functionality in Qt5. But it was deprecated/removed and never replaced. This port is one of the open-source replacements.

**Comment by dg0yt** [comment]
There was `QtSingleApplication`, but it was always living in some semi-official repositories.

**Pull request merged**